src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.schedulers;
+
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.MIReportingService;
+
+/**
+ * This scheduler checks if logical replication is in place.
+ */
+@Component
+public class LogicalReplicationCreatorScheduler {
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(LogicalReplicationCreatorScheduler.class);
+
+    @Autowired
+    private MIReportingService miReportingService;
+
+    @Scheduled(fixedDelay = 1000)
+    public void scheduled() {
+        LOGGER.debug("Postgresql logical replication scheduler executed");
+        miReportingService.logicalReplicationCheck();
```

Comment:
Will this check every second even if the previous check has blocked for a few seconds?
I am just thinking about what happens in that case?  Could you get 2 attempts to create a publication?  Does that matter or does it just work?

Reply:
Should be transactional

Reply:
Have run it on perftest and no problems